source/developer/webapp-component.rst

```diff
@@ -0,0 +1,180 @@
+Building a Webapp React Component
+==========================
+
+This page describes how to build a new React component in the webapp. All new components must meet the following requirements:
+
+1. Be pure, meaning that all information required to render are passed in by props
+2. Have no direct store interaction, use an index.js to wrap the component if needed
+3. Have component tests
+4. Be generic and re-usable when possible
+5. Have documented prop types
```

Comment:
Maybe change this to "Have documented props" since we want the purpose of the props to be clear